feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/encap_gre_ipv4_test.go

```diff
@@ -0,0 +1,1062 @@
+	ipv6Addrs := gnmi.LookupAll(t, dut, lo0.Ipv6().AddressAny().State())
+	if len(ipv4Addrs) == 0 && len(ipv6Addrs) == 0 {
+		loop1 := dutlo0Attrs.NewOCInterface(loopbackIntfName, dut)
+		loop1.Type = oc.IETFInterfaces_InterfaceType_softwareLoopback
+		gnmi.Update(t, dut, dc.Interface(loopbackIntfName).Config(), loop1)
+	} else {
+		v4, ok := ipv4Addrs[0].Val()
+		if ok {
+			dutlo0Attrs.IPv4 = v4.GetIp()
+		}
+		v6, ok := ipv6Addrs[0].Val()
+		if ok {
+			dutlo0Attrs.IPv6 = v6.GetIp()
+		}
+		t.Logf("Got DUT IPv4 loopback address: %v", dutlo0Attrs.IPv4)
+		t.Logf("Got DUT IPv6 loopback address: %v", dutlo0Attrs.IPv6)
+	}
+}
+
+func configureTcamProfile(t *testing.T, dut *ondatra.DUTDevice) {
+	if dut.Vendor() != ondatra.ARISTA || strings.ToLower(dut.Model()) == "ceos" {
+		t.Logf("Tcam profile not supported on %s %s", dut.Name(), dut.Model())
+		return
+	}
+
+	gnmiClient := dut.RawAPIs().GNMI(t)
+	t.Logf("Push the CLI Qos config:%s", dut.Vendor())
+	gpbSetRequest := buildCliSetRequest(tcamProfileConfig)
+	if _, err := gnmiClient.Set(context.Background(), gpbSetRequest); err != nil {
+		t.Fatalf("Failed to set tcam profile from cli: %v", err)
+	}
+}
+
+func configureStaticRoutes(t *testing.T, dut *ondatra.DUTDevice) {
+
+	configStaticRoute(t, dut, "192.0.2.0/30", "192.0.2.10", "0")
+	for index, dest := range tunnelDestinations {
+		configStaticRoute(t, dut, fmt.Sprintf("%s/32", dest), otgPort2.IPv4, strconv.Itoa(index+1))
+	}
+	configStaticRoute(t, dut, "2001:DB8:0::0/126", "2001:DB8:0::10", "0")
+}
+
+// Congigure Static Routes on DUT
+func configStaticRoute(t *testing.T, dut *ondatra.DUTDevice, prefix string, nexthop string, index string) {
+	b := &gnmi.SetBatch{}
+	if nexthop == "Null0" {
+		nexthop = "DROP"
+	}
+	routeCfg := &cfgplugins.StaticRouteCfg{
+		NetworkInstance: deviations.DefaultNetworkInstance(dut),
+		Prefix:          prefix,
+		NextHops: map[string]oc.NetworkInstance_Protocol_Static_NextHop_NextHop_Union{
+			index: oc.UnionString(nexthop),
+		},
+	}
+	if _, err := cfgplugins.NewStaticRouteCfg(b, routeCfg, dut); err != nil {
+		t.Fatalf("Failed to configure static route: %v", err)
+	}
+	b.Set(t, dut)
+}
+
+func buildCliSetRequest(config string) *gpb.SetRequest {
+	gpbSetRequest := &gpb.SetRequest{
+		Update: []*gpb.Update{
+			{
+				Path: &gpb.Path{
+					Origin: "cli",
+					Elem:   []*gpb.PathElem{},
+				},
+				Val: &gpb.TypedValue{
+					Value: &gpb.TypedValue_AsciiVal{
+						AsciiVal: config,
+					},
+				},
+			},
+		},
+	}
+	return gpbSetRequest
+}
+
+func runCliCommand(t *testing.T, dut *ondatra.DUTDevice, cliCommand string) string {
+	cliClient := dut.RawAPIs().CLI(t)
+	output, err := cliClient.RunCommand(context.Background(), cliCommand)
+	if err != nil {
+		t.Errorf("Failed to execute CLI command '%s': %v", cliCommand, err)
+	}
+	t.Logf("Received from cli: %s", output.Output())
+	return output.Output()
+}
+
+func configurePolicyForwarding(t *testing.T, dut *ondatra.DUTDevice) {
```

Comment:
Per the [contributing guide](https://github.com/openconfig/featureprofiles/blob/main/CONTRIBUTING.md#test-structure), this should be moved to internal/cfgplugins/policyforwarding.go

I suggest something like `func NewPolicyForwardingGreEncap`.  All the values should be in parameters (avoid hardcoding values) so it could be used by other tests doing GRE encap using policy-forwarding.

Deviations to the config generation should be inside the NewPolicyForwardingGreEncap function.

For style example, see https://github.com/openconfig/featureprofiles/blob/bc105a443b44d862c70e91112708b5a339c71ae5/internal/cfgplugins/sflow.go#L33